src/services/fluence-aurora-connector/modules/fluence_aurora_connector/src/main.rs

```diff
@@ -246,18 +280,113 @@ impl DealChangedResult {
 // `address` -- address of the deal we are modifying
 // `from_block` -- from which block to poll deals
 #[marine]
-pub fn poll_deal_change(net: String, address: String, from_block: String) -> DealChangedResult {
+pub fn poll_deal_changed(net: String, deal_id: String, from_block: String) -> DealChangedResult {
+    let address = format!("0x{}", deal_id);
     let to_block = get_to_block(&from_block);
-    let result = poll(net, address, from_block, to_block.clone(), DealChangedData::topic());
+    let result = poll(
+        net,
+        address,
+        from_block,
+        to_block.clone(),
+        DealChangedData::topic(),
+    );
     match result {
-        Err(err) => return DealChangedResult::error(err.to_string()),
+        Err(err) => return DealChangedResult::error(deal_id, err.to_string()),
         Ok(deals) => {
             let changed_deals = parse_deals::<DealChangedData, DealChanged>(deals);
-            DealChangedResult::ok(changed_deals, to_block)
+            DealChangedResult::ok(deal_id, changed_deals, to_block)
+        }
+    }
+}
+
+#[marine]
+pub struct DealUpdate {
+    deal_info: DealInfo,
+    from_block: String,
+}
+
+#[marine]
+pub struct DealInfo {
+    worker_id: String,
+    //spell_id: String,
+    deal_id: String,
+}
+
+#[marine]
+pub struct DealUpdatedBatchResult {
+    success: bool,
+    /// optional error
+    error: Vec<String>,
+    /// optional result (present if success is true)
+    result: Vec<DealChanged>,
+    /// The request checked blocks from `from_block` to `to_block`
+    to_block: String,
+    /// Return deal info to be able to find which deal to update
+    deal_info: DealInfo,
+}
+
+impl DealUpdatedBatchResult {
+    fn ok(to_block: String, deal_info: DealInfo, update: DealChanged) -> Self {
+        Self {
+            success: true,
+            error: vec![],
+            result: vec![update],
+            to_block,
+            deal_info,
+        }
+    }
+
+    fn error(to_block: String, deal_info: DealInfo, err: String) -> Self {
+        Self {
+            success: false,
+            error: vec![err],
+            result: vec![],
+            to_block,
+            deal_info,
         }
     }
 }
 
+#[marine]
+pub fn poll_deals_latest_update_batch(
+    net: String,
+    deals: Vec<DealUpdate>,
+) -> Vec<DealUpdatedBatchResult> {
+    let mut results = Vec::new();
+    for deal in deals {
+        let to_block = get_to_block(&deal.from_block);
+        let address = format!("0x{}", deal.deal_info.deal_id);
+        let result = poll(
+            net.clone(),
+            address,
+            deal.from_block,
+            to_block.clone(),
+            DealChangedData::topic(),
+        );
```

Comment:
Since the `net` is always the same, you should be able to batch several JsonRPC requests into a single `curl` invocation. 

Like that
```
curl -H "Content-Type: application/json" -X POST --data '[
    {"jsonrpc":"2.0","method":"web3_clientVersion","params":[],"id":67},
    {"jsonrpc":"2.0","method":"web3_clientVersion","params":[],"id":68}
]' https://...
```

https://www.jsonrpc.org/specification#batch